ikuzo/server.go

```diff
@@ -398,3 +396,9 @@ func (s *server) proxyDataNode(w http.ResponseWriter, r *http.Request) {
 
 	s.dataNodeProxy.ServeHTTP(w, r)
 }
+
+func (s *server) addShutdown(name string, hook Shutdown) {
+	if _, ok := s.shutdownHooks[name]; !ok {
```

Comment:
Does this check imply that it can be called multiple times for the same service?
If so, can not adding a shutdown hook if another hook for the same service is already in place cause issues?

Reply:
Each service can register shutdown hook. They register it by `name` and then it is called just once when the server shuts down.